src/transactions/TransactionFrame.cpp

```diff
@@ -818,6 +818,31 @@ TransactionFrame::processFeeSeqNum(AbstractLedgerTxn& ltx, int64_t baseFee)
     {
         throw std::runtime_error("Unexpected database state");
     }
+
+    if (protocolVersionStartsFrom(header.current().ledgerVersion,
+                                  ProtocolVersion::V_19))
+    {
+        auto ltxe = loadMaxSeqNumToApply(ltx, getSourceID());
+        if (!ltxe)
+        {
+            InternalLedgerEntry gle(
+                InternalLedgerEntryType::MAX_SEQ_NUM_TO_APPLY);
+            gle.maxSeqNumToApplyEntry().sourceAccount = getSourceID();
+            gle.maxSeqNumToApplyEntry().maxSeqNum = getSeqNum();
+
+            auto res = ltx.create(gle);
+            if (!res)
+            {
+                throw std::runtime_error("create failed");
+            }
+        }
+        else
+        {
+            ltxe.currentGeneralized().maxSeqNumToApplyEntry().maxSeqNum =
+                getSeqNum();
```

Comment:
This assumes `processFeeSeqNum` is called with increasing sequence numbers per source account. This is true at the moment, but it might be a good idea to use `std::max` or throw if the sequence numbers aren't increasing.

Reply:
I agree... at the same time, why not just use `max` now then as it makes it clear that this is the max over all txs for that source?

Reply:
Done